src/Entity/OgRole.php

```diff
@@ -269,6 +268,33 @@ public function isRequired() {
     return static::getRoleTypeByName($this->getName()) === OgRoleInterface::ROLE_TYPE_REQUIRED;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function isAdmin() {
+    return $this->getName() === OgRoleInterface::ADMINISTRATOR;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setIsAdmin($is_admin) {
+    parent::setIsAdmin($is_admin);
+
+    // This method has been inherited from the core RoleInterface but has the
+    // unexpected side effect that it will change the role name. The recommended
+    // way of defining the administrator role is by setting the role name to
+    // OgRoleInterface::ADMINISTRATOR, for example:
+    // @code
+    //   $role->setName(OgRoleInterface::ADMINISTRATOR);
+    // @endcode
+    trigger_error('Using ::setIsAdmin() affects the role name.', E_USER_WARNING);
+
+    $this->setName($is_admin ? OgRoleInterface::ADMINISTRATOR : OgRoleInterface::AUTHENTICATED);
```

Comment:
OK, that's also a good solution, alerting the developer using an exception.

@amitaibu to be 100% clear, it is the intention that a group admin has all permissions in the group, right? Because that is wat `::isAdmin()` promises:

```
  /**
   * Indicates that a role has all available permissions.
   *
   * @return bool
   *   TRUE if the role has all permissions.
   */
  public function isAdmin();
```

Reason I am asking is that in my project we have 1 test failing with this PR applied: we have a use case where the group admin is not allowed to delete content, because the content needs to be retained for archival reasons. With the PR the admin gets this permission.

So my understanding is that this is intentional, and on our end we should implement `hook_og_user_access_alter()` to take away the permission from the group admin?

Reply:
ahh, hello :) There is an explicit `is_admin` flag that controls this in core. We can use this. We don't need to restrict it in the administrator role.
The og_roles have inherited this flag and are already in exported configurations. Core also bases this in this flag.